File: flutter_hr_2/flutter_hr/lib/models/advance_salary.dart
class AdvanceSalary {
  final int? id;
  final int employeeId;
  final double amount;
  final String requestReason;
  final String? forMonth;
  final int? forYear;
  final int repaymentMonths;
  final double? monthlyDeduction;
  final String status; // Pending, Approved, Rejected, Paid, Partially Paid
  final String? requestDate;
  final String? approvalDate;
  final String? rejectionReason;
  final double? amountPaid;

  AdvanceSalary({
    this.id,
    required this.employeeId,
    required this.amount,
    required this.requestReason,
    this.forMonth,
    this.forYear,
    this.repaymentMonths = 1,
    this.monthlyDeduction,
    required this.status,
    this.requestDate,
    this.approvalDate,
    this.rejectionReason,
    this.amountPaid,
  });

  factory AdvanceSalary.fromJson(Map<String, dynamic> json) {
    return AdvanceSalary(
      id: (json['id'] as num?)?.toInt(),
      employeeId: (json['employeeId'] as num?)?.toInt() ?? 0,
      amount: (json['amount'] as num?)?.toDouble() ?? 0.0,
      requestReason: json['requestReason']?.toString() ?? json['reason']?.toString() ?? '',
      forMonth: json['forMonth']?.toString(),
      forYear: (json['forYear'] as num?)?.toInt(),
      repaymentMonths: (json['repaymentMonths'] as num?)?.toInt() ?? 1,
      monthlyDeduction: (json['monthlyDeduction'] as num?)?.toDouble(),
      status: json['status']?.toString() ?? 'Pending',
      requestDate: json['requestDate']?.toString(),
      approvalDate: json['approvalDate']?.toString(),
      rejectionReason: json['rejectionReason']?.toString(),
      amountPaid: (json['amountPaid'] as num?)?.toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'employeeId': employeeId,
      'amount': amount,
      'requestReason': requestReason,
      'forMonth': forMonth,
      'forYear': forYear,
      'repaymentMonths': repaymentMonths,
      if (monthlyDeduction != null) 'monthlyDeduction': monthlyDeduction,
      'status': status,
      if (requestDate != null) 'requestDate': requestDate,
      if (approvalDate != null) 'approvalDate': approvalDate,
      if (rejectionReason != null) 'rejectionReason': rejectionReason,
      if (amountPaid != null) 'amountPaid': amountPaid,
    };
  }
}

File: flutter_hr/HR-And-Payroll-Management-System-Backend/src/main/java/Israt/example/Hr/and/payroll/management/system/backend/entity/AdvanceSalary.java
package Israt.example.Hr.and.payroll.management.system.backend.entity;

import Israt.example.Hr.and.payroll.management.system.backend.enums.Month;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "advance_salaries")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AdvanceSalary {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "employee_id", nullable = false)
    private Long employeeId;

    @Column(nullable = false)
    private BigDecimal amount;

    @Column(length = 1000)
    private String requestReason;

    @Column
    private LocalDateTime requestDate = LocalDateTime.now();

    @Enumerated(EnumType.STRING)
    @Column(name = "for_month")
    private Month forMonth;

    @Column(name = "for_year")
    private Integer forYear;

    @Column(name = "repayment_months")
    private Integer repaymentMonths = 1;

    @Column(name = "monthly_deduction")
    private BigDecimal monthlyDeduction;

    @Enumerated(EnumType.STRING)
    @Column(name = "repayment_start_month")
    private Month repaymentStartMonth;

    @Column(name = "repayment_start_year")
    private Integer repaymentStartYear;

    @Column(length = 50)
    private String status; // Pending, Approved, Rejected, Paid, Partially Paid

    @Column(name = "approved_by")
    private Long approvedBy;

    @Column(name = "approval_date")
    private LocalDateTime approvalDate;

    @Column(length = 1000)
    private String rejectionReason;

    @Column(name = "rejected_by")
    private Long rejectedBy;

    @Column(name = "amount_paid")
    private BigDecimal amountPaid = BigDecimal.ZERO;

    @Column(length = 1000)
    private String attachments;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();

    @Column(name = "created_by")
    private Long createdBy;

    @Column(name = "updated_by")
    private Long updatedBy;

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}
